common/templates/record.html

```diff
@@ -15,9 +15,10 @@
             <span ng-if="showInlineTable($index)" id="entity-{{$index}}-table">
                 <record-action-bar id="actionbar-{{$index}}" is-inline="true" is-table-display="col.tableModel.isTableDisplay"
                     toggle-display-mode='toggleInlineDisplayMode({dataModel:col.tableModel})' can-edit="canEditRelated({ref:col.tableModel.reference}) && !showError($index)"
-                    can-create="canCreateRelated({ref:col.tableModel.reference})" show-toggle-display-btn="allowInlineTableMarkdown($index)"
+                    can-create="canCreateRelated({ref:col.tableModel.reference})" can-create-disabled="canCreateRelatedDisabled({ref:col.tableModel.reference})" show-toggle-display-btn="allowInlineTableMarkdown($index)"
                     add-related-record="addRelatedRecord({tableModel:col.tableModel})" to-recordset="toRecordSet({ref:col.tableModel.reference})"
-                    displayname="col.column.displayname" base-table-name="col.baseTableName">
+                    displayname="col.column.displayname" base-table-name="col.baseTableName"
+                    keyset="col.tableModel.reference.derivedAssociationReference ? col.tableModel.reference.derivedAssociationReference.origFKR.key.colset.columns : col.tableModel.reference.origFKR.key.colset.columns">
```

Comment:
You don't need to check for `derivedAssociationReference`. `origFKR` will be available in `col.tableModel.reference` in both one-hop related and p&b.

It won't be available in the free-form cases but given that we're not showing the "add" button in those cases, it shouldn't matter.